document API, extend tests a little (#1)
ssbc/ssb-tribes2: README.md

- Requires **Node.js 12** or higher
- Requires `secret-stack@^6.2.0`
- Requires `ssb-db2@>=5.0.0`
- Requires `ssb-box2@>2.0.2`
- TODO

```diff
 const ssb = SecretStack({ caps: require('ssb-caps') })
   .use(require('ssb-master'))
+  .use(require('ssb-db2'))
   .use(require('ssb-conn'))
+  .use(require('ssb-box2'))
+  .use(require('ssb-tribes2'))
   .use(require('ssb-blobs'))
   .call(null, config)

ssb.tribes2.start()

ssb.tribes2.create({}, (err, group) => {
  const content = {
    type: 'post',
    text: 'welcome to the group',
    recps: [group.id]
  }

>   ssb.db.publishAs(group.subfeed, content, cb)

Reviewer: We should use `ssb.db.create` not `ssb.db.publishAs`

Reviewer: ```suggestion
  const keys = group.mySubfeedKeys
  const recps = [group.id]
  const content = { type: 'post', text: 'welcome, friends!' }
  ssb.db.create({ keys, recps, content }, cb)
```

Author: What's the difference? I guess I don't fully understand why we need create - apart from being about to explicitly tell it to use box2 encryption?

Reviewer: Oh yeah, good point, we also need to add `encryptionFormat: 'box2'` to the opts.

The difference is that publishAs just calls `create` under the hood, and we'd (arj and I) would like to deprecate publish and publishAs in the near future. create() can do everything we need, and it can be misleading to use publish or publishAs because they don't properly support choosing feed formats and encryption formats.

Reviewer 2: @staltz done

Reviewer: @Powersource There are two `const content` definitions still in that readme

Reviewer: I did this final step

})
```

## API

### `ssb.tribes2.create(opts, cb)`

Creates a new private group.

> This creates an encruption key, sets up a sub-feed for the group, and initializes the

Reviewer: ```suggestion
This creates an encryption key, sets up a sub-feed for the group, and initializes the
```

group with a `group/init` message, and `group/add-member` to signal you were added.
Calls back with important info about the group

> - `opts` *Object* - currently empty, but will be used to specifiy details like whether the group has an admin subgroup etc. in future

Reviewer: ```suggestion
- `opts` *Object* - currently empty, but will be used in the future to specify details like whether the group has an admin subgroup, etc
```

> - `cb` *Function* - callback function of signature `(err, group)` where `group` in object containing:

Reviewer: ```suggestion
- `cb` *Function* - callback function of signature `(err, group)` where `group` is an object containing:
```

    - `id` *CloakedId* - a cipherlink that's safe to use publicly to name the group, and is used in `recps` to trigger encrypting messages to that group, encoded as an ssb-uri
    - `subfeed` *Keys* - the keys of the subfeed you should publish group data to

>     - `secret` *Buffer*  - the symmetric key used for encryption by the group

Reviewer: ```suggestion
    - `secret` *Buffer* - the symmetric key used by the group for encryption
```
